Sources/Swift/Metrics/LocalMetricsAggregator.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+
+/// Used for correlating metrics to spans. See https://github.com/getsentry/rfcs/blob/main/text/0123-metrics-correlation.md
+///
+class LocalMetricsAggregator {
+    
+    struct Metric {
+        let min: Double
+        let max: Double
+        let count: Int
+        let sum: Double
+        let tags: [String: String]
+    }
+    
+    private typealias ExportKey = String
+    private typealias TagsKey = String
+    
+    private var metricBuckets: [ExportKey: [TagsKey: Metric]] = [:]
+    private let lock = NSLock()
+    
+    func add(type: MetricType, key: String, value: Double, unit: MeasurementUnit, tags: [String: String]) {
+        let exportKey = "\(type.rawValue):\(key)@\(unit.unit)"
+        let tagsKey = getTagsKey(tags: tags)
+        
+        lock.synchronized {
+            var bucket = metricBuckets[exportKey] ?? [:]
+            
+            var metric = bucket[tagsKey] ?? Metric(min: value, max: value, count: 1, sum: value, tags: tags)
+            
+            if bucket[tagsKey] != nil {
+                let newMin = min(metric.min, value)
+                let newMax = max(metric.max, value)
+                let newSum = metric.sum + value
+                let count = metric.count + 1
+                
+                metric = Metric(min: newMin, max: newMax, count: count, sum: newSum, tags: tags)
+            }
+            
+            bucket[tagsKey] = metric
+            metricBuckets[exportKey] = bucket
+        }
+    }
+    
+    func serialize() -> [String: [[String: Any]]] {
+        var returnValue: [String: [[String: Any]]] = [:]
```

Comment:
m: Since you're declaring `typealias`, I think we should reuse it wherever we expect that type of data. The biggest benefit of typealias is to easily update types, and we don't have that here. For this, you need to make ExportKey not private.

Otherwise, using `typealias` in this context with a base type is not adding any value; on the contrary, it requires a second look to understand what's going on.

```suggestion
    func serialize() -> [ExportKey: [[String: Any]]] {
        var returnValue: [ExportKey: [[String: Any]]] = [:]
```